src/main/extension/extension_install.cpp

```diff
@@ -58,35 +58,36 @@ string ExtensionHelper::ExtensionInstallDocumentationLink(const string &extensio
 	return link;
 }
 
-duckdb::string ExtensionHelper::DefaultExtensionFolder(FileSystem &fs) {
+duckdb::pair<duckdb::string, idx_t> ExtensionHelper::DefaultExtensionFolder(FileSystem &fs) {
```

Comment:
Perhaps instead of tracking a "how_deep" variable, we can decompose the extension folder we return here:

```cpp
struct ExtensionFolder {
    string base_directory;
    vector<string> components;

    string GetPath(); // join the components together
};
```

Then we can call `CreateDirectoriesRecursive` on the components later on in the process.